lib/core/src/persist/receive.rs

```diff
@@ -83,13 +84,25 @@ impl Persister {
     }
 
     pub(crate) fn insert_or_update_receive_swap(&self, receive_swap: &ReceiveSwap) -> Result<()> {
+        let maybe_swap = self.fetch_receive_swap_by_id(&receive_swap.id)?;
+        let updated_fields = ReceiveSyncData::updated_fields(maybe_swap, receive_swap);
+
         let mut con = self.get_connection()?;
         let tx = con.transaction_with_behavior(TransactionBehavior::Immediate)?;
 
         Self::insert_or_update_receive_swap_inner(&tx, receive_swap)?;
-        self.commit_outgoing(&tx, &receive_swap.id, RecordType::Receive, None)?;
+
+        // Trigger a sync if:
+        // - updated_fields is None (swap is inserted, not updated)
+        // - updated_fields in a non empty list of updated fields
+        let trigger_sync = updated_fields.as_ref().map_or(true, |u| !u.is_empty());
+        self.commit_outgoing(&tx, &receive_swap.id, RecordType::Receive, updated_fields)?;
```

Comment:
The intent of the PR was to not commit or sync if we don't need to. It seems now we are committing, but not immediately syncing (will be synced in the next loop). I was hoping to avoid this by checking the updated fields to see if any relevant fields have changed.

Reply:
Sorry Ross, seems like a misunderstanding of https://github.com/breez/breez-sdk-liquid/pull/623#discussion_r1894859102. I think it makes sense to check the updated fields and them trigger the commit accordingly. I think it was referring to the final db transaction commit.